homework/homework1/task2/nl.py

```diff
@@ -0,0 +1,31 @@
+import sys
+from typing import TextIO
+
+
+def print_with_line_numbers(file_stream: TextIO, start_index: int):
+    index = start_index
+
+    for line in file_stream:
+        if line.strip():
+            print(f"{index} ", end="")
+            index += 1
+        print(line, end="")
+
+    return index
+
+
+def main(*files: str):
+    start_index = 1
+    for file in files:
+        try:
+            with open(file, "r") as f:
+                start_index = print_with_line_numbers(f, start_index)
```

Comment:
If you can, I would advise you to split this functionality. This function does two things:
1. prints the lines
2. returns the final value

This makes it difficult to read it: just look at this line, it's unobvious what `x = print...()` means.

Considering the fact that you are very good at using classes and structures in other places, I think you can do this more neatly.